### DISCUSSION TITLE
缓存清晰度最高只有720P

### DISCUSSION THREAD

**Discussion by dingdong624** [ywmoyue/biliuwp-lite#461] [bug]
### Bug 描述

缓存的优先清晰度只有720,480,360三个选项，不知道是因为什么

### 操作步骤

```plain text
下载
```

### 期望结果

希望能缓存最清晰的

### 截图

![哔哩哔哩 UWP 055646](https://github.com/ywmoyue/biliuwp-lite/assets/131755863/9071179c-55f3-4cd8-a24e-857682cbd6ca)

### 应用来源

Github/其他Github镜像站

### 代码贡献

我更擅长发现问题

**Comment by GD-Slime** [comment]
~请给出一下这个视频的bv号，这有助于发现bug根源。~
这个问题是由于下载时会请求分p中最后一个视频的清晰度并填入DownloaDialog, 而最后一个可能是防撞车的视频分辨率只有720p, 导致了此故障。

@ywmoyue 是否需要这样改：请求分p视频中每一个视频的视频信息，并找到最高的分辨率并填入DownloadDialog？ 
这样得修改下载时的行为，当检测到某个视频没有此分辨率时降一档分辨率并下载。

**Reply by ywmoyue** [reply]
感觉不妥，我晚点看看app是怎么处理的

**Reply by GD-Slime** [reply]
看了一下，大概B站客户端逻辑是这样的：
判断现在播放的是哪一p，并显示该p的所有可用分辨率。
如果此时点击“缓存全部”，则会下载该分辨率下所有视频。如果选择的分辨率高于某个分p，则会降到最高可用分辨率并下载该分p。

例如：在楼主这个例子中，最后一p的分辨率最高720p。
如果现在正在播放最后一p，那么只会加载最高720p的可用分辨率。点击缓存全部，则全部下载了720p视频。
如果正在播放第一p，会加载最高1080p的可用分辨率。点击全部下载，最后一p的清晰度则为720p。

**Reply by ywmoyue** [reply]
那就按b站客户端这样改吧，你这边改方便吗

**Reply by GD-Slime** [reply]
@ywmoyue 我正在改其他的东西，抱歉。如果您有时间就先改吧
